common/src/confidential/confidential_ctx.rs

```diff
@@ -188,10 +188,20 @@ impl EthConfidentialCtx for ConfidentialCtx {
     }
 
     fn encrypt_session(&mut self, data: Vec<u8>) -> Result<Vec<u8>> {
-        if self.peer_public_key.is_none() || self.contract.is_none() || self.next_nonce.is_none() {
-            return Err(Error::Confidential(
-                "must have key pair of a contract and peer and a next nonce".to_string(),
-            ));
+        if self.peer_public_key.is_none() {
+            err_msg.push_str("; no peer public key specified");
+        }
+        if self.contract.is_none() {
+            err_msg.push_str("; no contract specified from which to extract contract public key");
+        }
+        if self.next_nonce.is_none() {
+            err_msg.push_str("; no next nonce specified");
+        }
```

Comment:
can you push these to a vec and then `join("; ")` or else detect if the buffer is empty and not add `;`. I would feel demoralized to see `"; <error message>"`